src/containers/ToolboxSubject/ToolboxSubjectPage.tsx

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) 2021-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import React, { useEffect, useState } from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+// @ts-ignore
+import { OneColumn, ToolboxInfo, SubjectBanner } from '@ndla/ui';
+// @ts-ignore
+import { getUrnIdsFromProps, toTopic } from '../../routeHelpers';
+import { useGraphQuery } from '../../util/runQueries';
+import { subjectPageQuery } from '../../queries';
+import { DefaultErrorMessage } from '../../components/DefaultErrorMessage';
+import { GQLTopic, GQLSubject } from '../../graphqlTypes';
+import ToolboxTopicWrapper from './components/ToolboxTopicWrapper';
+import { LocaleType } from '../../interfaces';
+
+type Props = {
+  match: RouteComponentProps['match'];
+  locale: LocaleType;
+};
+
+interface Data {
+  subject: GQLSubject & { allTopics: GQLTopic[] };
+}
+
+const ToolboxSubjectPage = ({ match, locale }: Props) => {
+  const { subjectId, topicList } = getUrnIdsFromProps({
+    ndlaFilm: false,
+    match,
+  });
+
+  const refs = topicList.map(() => React.createRef<HTMLDivElement>());
+  const [selectedTopics, setSelectedTopics] = useState<GQLTopic[]>([]);
+  const { loading, data } = useGraphQuery<Data>(subjectPageQuery, {
+    variables: {
+      subjectId,
+    },
+  });
+
+  useEffect(() => {
+    topicList.forEach((topicId: string) => {
+      const alreadySelected = selectedTopics.find(
+        (topic: GQLTopic) => topic.id === topicId,
+      );
+      if (!alreadySelected) {
+        const exist = subject.allTopics.find(
+          (topic: GQLTopic) => topic.id === topicId,
+        );
+        if (exist) setSelectedTopics([exist, ...selectedTopics]);
+      }
+    });
+    scrollToTopic(topicList.length - 1);
+  });
+
+  const scrollToTopic = (index: number) => {
+    const ref = refs[index];
+    if (ref && ref.current) {
+      const positionFromTop =
+        ref.current.getBoundingClientRect().top +
+          document?.documentElement?.scrollTop || 100;
+      window.scrollTo({
+        top: positionFromTop - 100,
+        behavior: 'smooth',
+      });
+    }
+  };
+
+  if (loading) {
+    return null;
+  }
+
+  if (!data) {
+    return <DefaultErrorMessage />;
+  }
+
+  const subject = data.subject;
+
+  const topics = subject.topics?.map((topic: GQLTopic) => {
```

Comment:
Her så feiler TSC fordi typen `GQLSubject` har et field topics som er et array av topics. Dette arrayet med topics er blir definert som typen `Array<GQLTopic | null >`. Om man i graphql skriver typen til topics i subject queryen i scheme.ts til å være `topics: [Topic!]` så vil man istedenfor å få muligheten for `Array<GQLTopic | null>` få `Array<GQLTopic`. Denne typen vil så i gjengjeld tilslutt bare levere ett tomt array, fikk iallefall ingen tilfeller lokalt hvor den returnerte noen feilmeldinger.